### PR TITLE
Remove dependency on GAC

### PR DESCRIPTION
**Pull request by JoshVarty**
Now we almost exclusively rely on the output directory when resolving assemblies. (The exception might be `VSLoader` which I plan to remove next).

I've removed the postbuild step that was deleting all `Microsoft.VisualStudio.*` assemblies from the output directory.

I removed more `<SpecificVersion>False</SpecificVersion>` from VSEmbedRoslyn.csproj.

I removed the dependency on Program Files when resolving Roslyn references.

### CODE REVIEW

File: VSEmbed.DemoApp/VSEmbed.DemoApp.csproj [hunk 1]
```diff
@@ -41,6 +41,18 @@
       <HintPath>..\packages\BasicUndo.0.9.3\lib\net45\BasicUndo.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.VisualStudio.Editor.Implementation, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
```
Comment (AmadeusW) [inline VSEmbed.DemoApp/VSEmbed.DemoApp.csproj]: We usually remove specificversion=false, is it needed here?
It look like VS is adding this tag automatically

File: VSEmbed/VsMefContainerBuilder.cs [hunk 2]
```diff
@@ -54,28 +54,25 @@ private VsMefContainerBuilder(MEFv3.ComposableCatalog catalog)
...
+				"Microsoft.CodeAnalysis.VisualBasic.Workspaces",
+				"Microsoft.CodeAnalysis.Workspaces.Desktop",
+				"Microsoft.CodeAnalysis.Workspaces",
+
+				"Microsoft.VisualStudio.Platform.VSEditor",
+				"Microsoft.VisualStudio.Text.Logic",
+				"Microsoft.VisualStudio.Text.UI",
+				"Microsoft.VisualStudio.Text.UI.Wpf",
+				"Microsoft.VisualStudio.Editor.Implementation",
+				"Microsoft.VisualStudio.Shell.TreeNavigation.HierarchyProvider",
```
Comment (AmadeusW) [inline VSEmbed/VsMefContainerBuilder.cs:71]: Nice :)

**Review by AmadeusW — COMMENTED** [review]
specificversion issue

File: VSEmbed.DemoApp/VSEmbed.DemoApp.csproj (hunk 1 above)
Comment (JoshVarty) [reply VSEmbed.DemoApp/VSEmbed.DemoApp.csproj]: Weird, I don't think it should be there...

**Pull request merged**